shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -272,16 +297,131 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool) extend
           td.nme.name -> td :: Nil
         else Nil
       })
+    } |> { implicit ctx =>
```

Comment:
Okay, my original code was already borderline with its use of `} |> { implicit ctx =>`, but this is getting out of hands 😅 
We should split all these typing phases into properly-named local `def`s, ideally with one or two lines of comment each documenting what it's doing.

Reply:
Addressed in f51f6f231a75d9ca2f52f1485b37b99d580eaaac